portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,10 +48,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<String> comments = new ArrayList<String>();
+    HashMap<String, ArrayList<String>> comments = new HashMap<String, ArrayList<String>>();
     for (Entity entity : results.asIterable()) {
-      String comment = (String) entity.getProperty("comment");
-      comments.add(comment);
+      String id = KeyFactory.keyToString(entity.getKey());
+      ArrayList<String> commentEmail = new ArrayList<String>();
```

Comment:
We typically use the interface for the type of collection variables, i.e.

List<String> commentEmail = new ArrayList<String>();   // uses interface as type
ArrayList<String> commentEmail = new ArrayList<String>();    // uses class as type

Reply:
changed this one and also when creating the Hashmap